pkg/agent/ebpfspy/symbols_linux.go

```diff
@@ -0,0 +1,86 @@
+//go:build ebpfspy
+// +build ebpfspy
+
+// Package ebpfspy provides integration with Linux eBPF. It is a rough copy of profile.py from BCC tools:
+//   https://github.com/iovisor/bcc/blob/master/tools/profile.py
+package ebpfspy
+
+import (
+	"debug/elf"
+	"unsafe"
+)
+
+/*
+#include "bcc_syms/bcc_syms.h"
+*/
+import "C"
+
+var globalCache *symbolCache
+
+func init() {
+	globalCache = newSymbolCache()
+}
+
+type symbolCache struct {
+	cachePerPid map[uint32]unsafe.Pointer
+}
+
+func newSymbolCache() *symbolCache {
+
+	return &symbolCache{
+		cachePerPid: make(map[uint32]unsafe.Pointer),
+	}
+}
+
+func (sc *symbolCache) cache(pid uint32) unsafe.Pointer {
+	if cache, ok := sc.cachePerPid[pid]; ok {
+		return cache
+	}
+	pidC := C.int(pid)
+	if pid == 0 {
+		pidC = C.int(-1)
+	}
+	symbolOpt := C.struct_bcc_symbol_option{use_symbol_type: C.uint(1 << elf.STT_FUNC)}
+	symbolOptC := (*C.struct_bcc_symbol_option)(unsafe.Pointer(&symbolOpt))
+	cache := C.bcc_symcache_new(pidC, symbolOptC)
```

Comment:
I wonder how this ownership is managed. Shouldn't we free it explicitly with [bcc_free_symcache](https://github.com/iovisor/bcc/blob/f6c4b4c618a1a471fdfdd8bc852578f383793b63/src/cc/bcc_syms.h#L57-L58) at some point and delete the entry from Go map?

Reply:
I agree we should call `bcc_free_symcache` and delete it from the map
- when the ebpfspy is stopping
- when pid is dead
- maybe even add some lru cache for some 
I suspect this unfreed cache may be the cause of todays memory consumption spike

Reply:
An LRU is great idea, I like it